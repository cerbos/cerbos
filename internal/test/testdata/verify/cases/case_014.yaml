description: Test suite with invalid effects
