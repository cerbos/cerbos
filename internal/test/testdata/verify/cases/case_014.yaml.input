-- testdata/principals.yaml --
---
principals:
  john:
    id: john
    policyVersion: '20210210'
    roles:
      - employee
    attr:
      department: marketing
      geography: GB
      team: design
  bev: &bev
    id: bev
    policyVersion: '20210210'
    roles:
      - employee
      - manager
    attr: &bev_attr
      department: marketing
      geography: GB
      managed_geographies: GB
      ip_address: 10.20.1.2
      team: design
  matt:
    << : *bev
    id: matt
    attr:
      << : *bev_attr
      ip_address: 10.10.1.2

-- testdata/resources.yaml --
---
resources:
  john_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX125
    attr: &attr
      department: marketing
      geography: GB
      id: XX125
      owner: john
      team: design
  pending_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX125
    attr:
      << : *attr
      status: PENDING_APPROVAL
  stale_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX225
    attr:
      << : *attr
      modifiedAt: "2022-08-01T15:00:00Z"
  stale_pending_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX225
    attr:
      << : *attr
      modifiedAt: "2022-08-01T15:00:00Z"
      status: PENDING_APPROVAL

-- testdata/auxdata.yaml --
---
auxData:
  myJWT:
    jwt:
      iss: cerbos-test-suite
      aud: [cerbos-jwt-tests]
      customArray: [A, B]

-- suite_test.yaml --
---
name: TestSuite
description: Tests for verifying something
tests:
  - name: John and his leave request
    input:
      principals:
        - john
      resources:
        - john_leave_request
      actions:
        - view:public
        - approve
        - defer
      auxData: myJWT
    expected:
      - principal: john
        resource: john_leave_request
        actions:
          view:public: EFFECT_UNSPECIFIED
          approve: EFFECT_NO_MATCH
          defer: EFFECT_ALLOW

-- another_suite_test.yaml --
---
name: AnotherTestSuite
description: Tests for verifying something
tests:
  - name: John and his leave request
    input:
      principals:
        - john
      resources:
        - john_leave_request
      actions:
        - approve
        - defer
      auxData: myJWT
    expected:
      - principal: john
        resource: john_leave_request
        actions:
          approve: EFFECT_ALLOW
          defer: EFFECT_ALLOW
          reject: EFFECT_ALLOW

