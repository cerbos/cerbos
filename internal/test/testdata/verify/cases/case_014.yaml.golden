{
  "suites": [
    {
      "file": "another_suite_test.yaml",
      "name": "AnotherTestSuite",
      "summary": {
        "overallResult": "RESULT_ERRORED"
      },
      "error": "Failed to load the test suite: invalid test \"John and his leave request\": found expectations for actions that do not exist in the input actions list: [reject]",
      "description": "Tests for verifying something"
    },
    {
      "file": "suite_test.yaml",
      "name": "Unknown",
      "summary": {
        "overallResult": "RESULT_ERRORED"
      },
      "error": "file is not valid: [/tests/0/expected/0/actions/view:public: value must be one of \"EFFECT_ALLOW\", \"EFFECT_DENY\", /tests/0/expected/0/actions/approve: value must be one of \"EFFECT_ALLOW\", \"EFFECT_DENY\"]"
    }
  ],
  "summary": {
    "overallResult": "RESULT_ERRORED"
  }
}
